toolchain/check/context.cpp

```diff
@@ -251,6 +251,31 @@ auto Context::is_current_position_reachable() -> bool {
          SemIR::TerminatorKind::Terminator;
 }
 
+namespace {
+class CopyOnWriteBlock {
```

Comment:
Some comments to document how this is supposed to work?

Seems especially important as C-o-W semantics can be a bit subtle.